llama-index-integrations/llms/llama-index-llms-sagemaker-endpoint/llama_index/llms/sagemaker_endpoint/base.py

```diff
@@ -83,7 +83,7 @@ def __init__(
         aws_access_key_id: Optional[str] = None,
         aws_secret_access_key: Optional[str] = None,
         aws_session_token: Optional[str] = None,
-        region_name: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
```

Comment:
a tad weary of changing a core interface kwarg name. We might have to keep both and mark the old name as deprecated?

Reply:
@logan-markewich my rationale was to preserve consistency with the [documentation](https://docs.llamaindex.ai/en/stable/examples/llm/sagemaker_endpoint_llm.html). Anyone using sagemaker with llamaindex would have had to look at the source code and manually pass `region_name` as an kwarg (which is what we did and I was the first to raise the issue (#11999) alongside the PR). In my view adding extra code to pass `aws_region_name` by default and fall back to `region_name` if `aws_region_name` is `None` would also impact readability, but can do so if that's the direction you prefer.

Reply:
This is a breaking change to a very public interface -- I would prefer to give users some period of time to update their code. If they pass in the old kwarg, we should raise a warning that its deprecated

Reply:
another option is to pass this as a `kwargs` and make the old parameter as deprecated